### ISSUE TITLE
feat(language): add embedding layer and vector space

### ISSUE THREAD

**Issue by leandronsp** [leandronsp/aspirina#3]
## Overview

Implement an embedding layer in `language/embedding.rs` that converts token IDs into dense vectors. Add softmax to `core/calc.rs`. This bridges tokenization and the neural network: tokens become geometry.

## Problem Statement

The tokenizer produces integer IDs. The transformer needs dense vectors it can do math on. An embedding layer is a lookup table: each token ID maps to a learnable vector of N dimensions. These vectors capture semantic meaning after training. Tokens with similar meaning end up close in vector space.

## Current State

- Tokenizer exists in `language/tokenizer.rs` (from issue #2), producing `Vec<u32>`
- `core/calc.rs` has sigmoid and tanh but no softmax
- `core/matrix.rs` handles all matrix operations

## Requirements

- [ ] R-1: Implement `Embedding` struct: a matrix of shape `(vocab_size, embed_dim)` initialized with small random values
- [ ] R-2: Forward pass: given `Vec<u32>` token IDs, return a `Matrix` of shape `(seq_len, embed_dim)`
- [ ] R-3: Backward pass: given gradients of shape `(seq_len, embed_dim)`, accumulate gradients for the looked-up rows
- [ ] R-4: Add `softmax` function to `core/calc.rs` with derivative
- [ ] R-5: Unit tests for embedding lookup, gradient accumulation, and softmax correctness
- [ ] R-6: Demonstrate: tokenize a sentence, embed it, show that vectors exist in a space where distances are meaningful (even if random before training)

## Technical Approach

```
language/src/
  embedding.rs    # Embedding struct with forward(), backward()

core/src/
  calc.rs         # add softmax(), softmax_derivative()
```

Embedding is NOT a matrix multiply. It's a table lookup (index into rows). This is why `layer.rs` from gates doesn't apply here.

## Acceptance Criteria

- Given vocab_size=1000 and embed_dim=64, when I create an Embedding, then the internal matrix has shape (1000, 64)
- Given token IDs [42, 7, 13], when I call forward(), then I get a Matrix of shape (3, 64)
- Given softmax([1.0, 2.0, 3.0]), when I compute it, then the output sums to 1.0

## Out of Scope

- Positional encoding (comes with the transformer block)
- Pre-trained embeddings. Everything trains from scratch
- Visualization tools for vector space
